### PR TITLE
Reimplement Spinner component without Shoelace

### PR DESCRIPTION
**Pull request by oyamauchi**
## Description

This mimics the MUI `CircularProgress` that we use on the consumer
site. It looks different from the existing Shoelace spinner (which has
an always-visible circular track, and uses different colors), but that
was just a convenient off-the-shelf thing, not a considered design
choice.

## Test Plan

Look at it in its two contexts (within the utility selector, and while
loading incentive results). Helps to turn on throttling in web
inspector so it's there for longer.

### CODE REVIEW

File: src/components/spinner.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    role="progressbar"
+  >
+    <svg className="w-full h-full" viewBox="22 22 44 44">
+      <circle
+        className="animate-spinnerIndicator"
+        cx="44"
+        cy="44"
+        r="20.2"
+        fill="none"
+        stroke="currentColor"
```
Comment (veekas) [inline src/components/spinner.tsx:16]: huh i don't think i knew about `currentcolor`!

**Review by veekas — APPROVED** [review]
I like it! A few notes:

- is there a UX guidance for this? I'm curious about whether the spinner colors are correct, for example. I also feel like the spinner is potentially a bit too close to the dropdown arrow. but I'd of course defer to Emiola on both

- on the latter topic, I'm wondering if we should remove the caret when the dropdown is in a loading state. or, to be more analogous to PEP, add the spinner in front of the label text, rather than following it

Both aren't blockers

File: src/components/spinner.tsx (hunk 1 above)
Comment (oyamauchi) [reply src/components/spinner.tsx:16]: It becomes whatever the text color is at this point in the DOM, if the SVG is part of the DOM. (If the SVG is loaded via an img tag, currentColor is just black.)

**Review by oyamauchi — COMMENTED** [review]
The look was taken from the "loading" checkbox here: https://storybook.rewiringamerica.org/?path=/docs/inputs-checkbox--docs

Though I do need to double-check the color is right.

The positioning of the spinner is "that doesn't look terrible" on my part; I'm open to design feedback on that.

I've tried putting the spinner to the left of the label in the select, and it makes the text jump around in an unpleasant way when it appears/disappears. And preventing that by leaving space for the spinner on the left looks bad when the spinner is not there.

**Pull request merged**